### PR TITLE
feat: Climb/Route coordinates

### PR DESCRIPTION
**Pull request by alexoj46**
---
name: feat: Climb/Route coordinates

---

## What type of PR is this?(check all applicable)
- [ ] refactor
- [X] feature
- [ ] bug fix
- [ ] documentation
- [ ] optimization
- [ ] other

## Description

Implemented FrontEnd solution to Issue #1078 to add climb/route coordinates to climb pages.

### Related Issues

Issue #1078 

### What this PR achieves

Added FrontEnd component to climb/route page to display route coordinates. The component will also link to Google Maps, and will add a tick to the map on the website.

### Screenshots, recordings

Screenshot 2024-02-08 at 7 45 40 PM

### Notes
If coordinates are not present in metadata for that climb, the lat, lng display will still appear, with NA, NA as the value (as shown in the photo).

Also added my contributor information.

### CODE REVIEW

**Comment by btmccord** [comment]
Rather than showing NA, we should inherit the parents coordinates.

**Comment by musoke** [comment]
See https://github.com/OpenBeta/open-tacos/issues/1057 for discussion of
why inheriting parent coordinates may not be the best approach.

**Comment by btmccord** [comment]
> See #1057 for discussion of why inheriting parent coordinates may not be the best approach.

That makes sense for areas, however if we have a coordinate for an area we should show it on the route page as a very high percentage of the time they will be that same (for our purposes as climbers).

Route coordinates are only really usefully for a handful of areas that are very low density.

What about indicating that the coordinates displayed are that of the area?

**Comment by musoke** [comment]
Yes, that would probably do the trick - maybe even just show them grayed
out?

File: src/assets/contributors-map.geojson [hunk 1]
```diff
@@ -136,6 +136,21 @@
...
+      "properties": {
+        "firstName": "Alex",
+        "githubId": "alexoj46",
+        "favoriteCrag": "b81f8942-4e04-56eb-a681-84568676c681"
+      },
+      "geometry": {
+        "type": "Point",
+        "coordinates": [
+          34.18008,
+          -85.81680
```
Comment (vnugent) [inline src/assets/contributors-map.geojson:150]: Can you swap the order? Coordinate pair is expressed as longitude/lat (x,y) in geojson

**Comment by vnugent** [comment]
I've just looked at the backend code.  When creating a new route, the backend populates the climb's coordinate pair with that of the parent.  We can hash out more detail in #1057.  For this PR we can just deal with the visual.  If the climb object doesn't have coordinates, it's not easy to fetch the parent's nor it is worth it to deal with in the frontend (I think the backend is a better place for it).  Can we just not show it if that's the case?

What about adding some wording to indicate the coordinates are inherited form the parent? Something like `34.18150, -85.81285 (Inherited)`?

---

Screenshot 2024-02-11 at 10 40 40 AM

**Comment by btmccord** [comment]
> When creating a new route, the backend populates the climb's coordinate pair with that of the parent.

If this is the case I would think it's not possible to tell if it's inherited or not...?

I guess that right now they are all inherited since we have now editing functionality for climb coordinates currently?

**Comment by vnugent** [comment]
> I guess that right now they are all inherited since we have now editing functionality for climb coordinates currently?

Right now they all inherited, but we can't edit climb coordinates at the moment.

**Review by vnugent — CHANGES_REQUESTED** [review]

**Pull request closed without merging**